examples/aa-simple-dapp/src/hooks/usePlugin.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  UninstallPluginParams,
+  accountLoupeDecorators,
+  pluginManagerDecorator,
+  type IMSCA,
+  type Plugin
+} from "@alchemy/aa-accounts";
+import { AlchemyProvider } from "@alchemy/aa-alchemy";
+import { Address, UserOperationOverrides } from "@alchemy/aa-core";
+import { useCallback, useMemo } from "react";
+
+export const usePlugin = <AD, PD>(
+  provider: AlchemyProvider,
+  plugin: Plugin<AD, PD>,
+  pluginAddress_?: Address,
+) => {
+  if (!provider.isConnected<IMSCA>()) {
+    throw new Error(
+      "Provider must be connected to an MSCA before using this hook!"
+    );
```

Comment:
we should not throw, instead just return the null/empty defaults

Reply:
ah yea good call... I'll work on the typing and see if I can make it a bit easier for the consumer too